### DISCUSSION TITLE
Direct system calls for the foreign system

### DISCUSSION THREAD

**Discussion by gingerBill** [odin-lang/Odin#968] [Ideas/Requests]
Direct system calls for the foreign system.

**Comment by ThisDevDane** [comment]
Please confirm your request to the [issue template](https://github.com/odin-lang/Odin/blob/master/.github/ISSUE_TEMPLATE/feature_request.md)

**Comment by gingerBill** [comment]
@ThisDrunkDane This is more of a TODO for me rather than an actual request. I am being lazy, I know.

**Comment by dotbmp** [comment]
What would this look like? Literally using an integer value to define a syscall mapping?

Also, you could use projects to TODO yourself 😛

**Comment by gingerBill** [comment]
@bpunsky I keep forgetting projects exist...

And yes. It would pretty much be an extension of the foreign system. e.g. something like this:
```
foreign {
    @(system_call_id=156)
    foo_bar :: proc(x: uintptr, y: uintptr, z: ^i32) -> u64 ---
}
```
